### PR TITLE
Add capability to change the stream's resolution from application side

### PR DESCRIPTION
**Pull request by uniederer**
After some comment by @sabitertan in #5 I wanted to improve the component such that it allows the application to control the stream's resolution nearly live. 

I was considering to call applyConstraints on the track, yet it didn't give a smooth transition either. So I sticked to a complete restart in order to avoid potential side effects induced by sudden stream resolution changes.

### CODE REVIEW

**Comment by sabitertan** [comment]
Ideally we shouldn't create side effects in JSInterop. And I am not sure restarting stream is ideal for library too.

**Comment by uniederer** [comment]
> Ideally we shouldn't create side effects in JSInterop. 

What side effect are you referring to?

> And I am not sure restarting stream is ideal for library too.

Well at least it is of the same effect as if the user hit the start/stop buttons. What crossed my mind though was, whether it's worth to recreate the BarcodeReader in JS every time we start the stream to "reset" the whole thing. On the other hand, `decodeFromConstraints` in zxing-js' `BrowserCodeReader` calls through some hops `decodeFromStream` which in turn resets calls `this.reset()` (which stops camera, etc.) and then `this.attachStreamToVideo`.

So I'm not very concerned about that approach.

File: BlazorBarcodeScanner.ZXing.JS/JsInteropClass.cs [hunk 1]
```diff
@@ -1,73 +1,84 @@
...
+        {
+            // Implemented in BlazorBarcodeScannerJsInterop.js
+
+            return jsRuntime.InvokeAsync<List<VideoInputDevice>>(
+                "BlazorBarcodeScanner.listVideoInputDevices",
+                message);
+        }
+
+        public static void StartDecoding(IJSRuntime jSRuntime, string videoElementId, int width, int height) {
+            SetVideoResolution(jSRuntime, width, height);
```
Comment (sabitertan) [inline BlazorBarcodeScanner.ZXing.JS/JsInteropClass.cs:20]: I think we should move this out of JSInterop, either in Blazor component or into JS.

Comment (uniederer) [reply BlazorBarcodeScanner.ZXing.JS/JsInteropClass.cs:20]: I wanted to keep the interface that passes the actual interop layer (i.e. the JS methods called) as lean as possible. Hence I decided against having an extra method in the JS part.

However, on the other hand according to the single responsibility principle, I see little value in distributing the same functionality (`StartDecoding`) among multiple classes, as `StartDecoding(IJSRuntime jSRuntime, string videoElementId, int width, int height)` is just decorating (overloading) `StartDecoding(IJSRuntime jSRuntime, string videoElementId)`. 

At the end of the day, in my eyes, the overloading method is sort of "syntactical sugar" for the client of `JsInteropClass`.

Comment (sabitertan) [reply BlazorBarcodeScanner.ZXing.JS/JsInteropClass.cs:20]: Ah, now I got your intention.
Looks like you are working on another branch too.
Can you pls finalize PR7 so we can merge.
Thanks!

Comment (uniederer) [reply BlazorBarcodeScanner.ZXing.JS/JsInteropClass.cs:20]: Sure. I'll combine all the changes in #7 . Originally I wanted to have the incrments a bit more atomic, but I just found that the changes will interfere later on anyway so it makes sense to me to have all at once. Sorry for the confusion :-D.

**Comment by uniederer** [comment]
These changes will go along with those proposed in #7 .

**Pull request closed without merging**